File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/Divide.java
// Divide.java

public class Divide extends Binary
{
	public Divide(Expression aOperand1, Expression aOperand2)
	{
		super(aOperand1, aOperand2);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/solution/divide.cpp
// divide.cpp

#include "binary.h"
#include "divide.h"
class expression;
#include "expression_visitor.h"

divide::divide(expression * a_operand1, expression * a_operand2)
: binary(a_operand1, a_operand2)
{
}

void divide::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}
